zhinst/hdiq/hdiq.py

```diff
@@ -109,7 +109,7 @@ def _send_request(self, request: str) -> str:
         Returns:
             str: Returns the corresponsing status
         """
-        return self._send_command_loop(request, is_request=True)
+        return str(self._send_command_loop(request, is_request=True))
```

Comment:
Actually, the problem is deeper. `_send_command` returns `str` in the happy case, but `None` in case of a timeout. This cast to `str` silences the linter, but turns `None` into the result string `"None"`, which seems like a bad idea, as it is then impossible to distinguish a valid response from the error case.

For pragmatic reasons, I'd suggest to return an empty string in case of timeout, so:
```python
reply = self._send_command_loop(request, is_request=True)
return reply if reply is not None else ""
```

The casts to `bool` from above seem unnecessary in comparison, since `ack` in `_send_command` is boolean for all execution paths.

Reply:
Good point about the `None`. Initially I changed the `result` in `_send_command` default to an empty string but that might have other side effects. I will change according to your proposal.

Regarding the casts to bool: If I don't cast I get linter warnings:

![image](https://user-images.githubusercontent.com/1519902/224300619-c6800d57-aa75-43be-9548-b6d610c6ab0c.png)

But that might not be mypy complaining.

Reply:
I realized I left the `str` cast and wanted to remove it but this leads to a follow-up lint issue. I'm looking at it.